src/client/java/net/neoforged/neoforge/client/event/RegisterMetadataSectionTypesEvent.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import java.util.function.Consumer;
+import net.minecraft.server.packs.metadata.MetadataSectionType;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.fml.event.IModBusEvent;
+import org.jetbrains.annotations.ApiStatus.Internal;
+
+/**
+ * Fired to allow mods to register their own {@linkplain MetadataSectionType metadata section types}.
+ * This event is fired once on startup during the initial resource reload.
+ *
+ * <p>This event is not {@linkplain ICancellableEvent cancellable}.</p>
+ *
+ * <p>This event is fired on the mod-specific event bus, only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
+ */
+public class RegisterMetadataSectionTypesEvent extends Event implements IModBusEvent {
```

Comment:
This is specific to sprite loading. This should be explained in the javadoc and hinted at in the event name.

Reply:
I agree that it should be documented, but the reason why I chose a more generic name and didn't document it was because I was considering the use of this event for the other areas in sprite loading (like TextureAtlasHolder). As it stands right now, this event only affects a subset of atlases (namely, those using the `DEFAULT_METADATA_SECTIONS` set, which include vanilla atlases, inheritors of TextureAtlasHolder using the three-argument constructor, and people manually stitching atlases with the three-argument overload of `SpriteLoader#loadAndStitch`) and I wasn't entirely sure if there were any additional scenarios I was missing.

Reply:
It should for sure mention sprite or atlas I think.

Reply:
I renamed the class as well as added a whole chunk of documentation about the limitations of the event